### PR TITLE
[IMP] pos_reprint: make reprint also available without proxy print

### PR DESCRIPTION
**Pull request by wtaferner**
**Description of the issue/feature this PR addresses:**
PoS Reprint was only available for proxy printing which is not ok imho ;-)

**Current behavior before PR:**
No reprinting of tickets possible if the customer was using local printers

**Desired behavior after PR is merged:**
Now we introduce an own screen for reprinting and reuse a slightly changed receipt screen to make both scenarios possible.

@jorenvo @mart-e 
May I ask you to review and consider merging in stable? The improved version of this (maybe for current master?) should have the possibility to choose the order you want to reprint and save the html + xml in the backend for archiving these tickets.

Honestly, I do not like the XML print as it is not really flexible and ugly. But yeah, it is the only way to empower tablets and other devices to print via Posbox in case we do not allow to render the same view as rendered in JavaScript directly on the Posbox.

Thank you.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/pos_reprint/static/src/js/reprint.js [hunk 1]
```diff
@@ -14,11 +16,140 @@ devices.ProxyDevice.include({
     },
 });
 
+screens.ReceiptScreenWidget.include({
```
Comment (wtaferner) [inline addons/pos_reprint/static/src/js/reprint.js]: We might avoid this if we split the rendering of QWeb and DOM manipulation in ```point of sale```module

**Comment by wtaferner** [comment]
@mart-e @jorenvo 
Not interested at all or just had no time to give a statement or opinion?

**Comment by mart-e** [comment]
No review yet, sorry 😉 
Need to double check the impact, for instance we have to disable the reprint of VAT tickets for Belgian certification.

**Comment by wtaferner** [comment]
I see, right, certification is crucial. I resolved the reprint issue with taking the historical structured string and injected text to clearly show that it is a reprint, but I don't know how it was resolved in Belgium...I will probably take a look at the module to get an idea to know what is important in Belgium next time to include this in my future fixes (can't be to difficult)

**Comment by tivisse** [comment]
@jorenvo I think that you're the most qualified to look at this PR. 

Thanks.

**Comment by jorenvo** [comment]
Hi Wolfgang, I'll review this. But I'm already telling you that I don't want to merge this in a stable release. It's pretty much a new feature and as Martin mentioned we have to worry about things like our Belgian certification etc. So this will most likely go in master.

**Comment by jorenvo** [comment]
I wanted to try to avoid the code duplication so I came up with https://github.com/odoo/odoo/pull/17504.

**Comment by jorenvo** [comment]
Closed in 4d461cbe4170e8d5dbdb0c6bef1b313982d064fd.

**Pull request closed without merging**